File: Type/ModelManager.go
package Type

import (
	"encoding/json"
	"io/ioutil"
	"log"
)

type (
	// ModelManager ...
	ModelManager struct {
		List map[string]*MobModel
	}
	MobModel struct {
		Name     string            `json:"name"`
		Type     string            `json:"type"`
		Geom     string            `json:"geom"`
		BaseSkin string            `json:"baseSkin"`
		Skins    map[string]string `json:"skins"`
	}
)

//
func (this *ModelManager) Init() {
	log.Println("ModelManager.Init()")
	//
	this.List = make(map[string]*MobModel)
	//
	file, err := ioutil.ReadFile("www/media/models/database.json")
	if err != nil {
		log.Fatal(err)
	}
	data := make(map[string]interface{})
	_ = json.Unmarshal([]byte(file), &data)
	for key, value := range data {
		// Read assets
		if key == "assets" {
			for model_name, asset := range value.(map[string]interface{}) {
				asset_bytes, err := json.Marshal(asset)
				if err == nil {
					model := &MobModel{}
					err := json.Unmarshal(asset_bytes, model)
					if err == nil {
						model.Name = model_name
						this.List[model_name] = model
					}
				}
			}
		}
	}
}

File: node_server/model_manager.ts
import {Resources} from "@client/resources.js";

export class ModelManager {
    list: Map<string, any>;

    constructor() {}
    
    async init() {
        console.debug('ModelManager.Init()');
        this.list = new Map();
        //
        // const data = await Resources.loadModels()
        // for(const key in data) {
        //     // Read assets
        //     if (key == 'assets') {
        //         const models = data[key];
        //         for(const model_name in models) {
        //             const model = models[model_name];
        //             model.name = model_name;
        //             this.list.set(model_name, model);
        //         }
        //     }
        // }
    }

}
